packages/react/src/side-nav/SideNavLinkList.tsx

```diff
@@ -1,51 +1,118 @@
 	activePath: string | undefined;
 	items: SideNavProps['items'];
+	subLevelVisible: SideNavProps['subLevelVisible'];
 };
 
-export function SideNavLinkList({
-	activePath,
-	items: itemsProp,
-}: SideNavLinkListProps) {
-	const items = useMemo(() => {
-		return itemsProp.map((item) => ({
+type ItemWithIsActive = Omit<SideNavLinkListProps['items'][number], 'items'> & {
+	isActive?: boolean;
+	items?: ItemWithIsActive[];
+};
+
+// Recursively add `isActive` to any sub-level items
+const addIsActive =
+	(activePath: SideNavLinkListProps['activePath']) =>
+	(item: ItemWithIsActive): ItemWithIsActive => {
+		const isCurrentPage = item.href === activePath;
+		const isActive =
+			isCurrentPage || hasSubLevelActiveItem(item.items, activePath);
+
+		return {
 			...item,
-			isActive:
-				item.href === activePath || hasNestedActiveItem(item.items, activePath),
-		}));
-	}, [activePath, itemsProp]);
+			isActive,
+			items:
+				isActive || item.items?.length
+					? item.items?.map(addIsActive(activePath))
+					: item.items,
+		};
+	};
+
+export const SideNavLinkList = ({
+	activePath,
+	items,
+	subLevelVisible,
+}: SideNavLinkListProps) => {
+	const itemsWithIsActive = useMemo(() => {
+		return items.map(addIsActive(activePath));
+	}, [activePath, items]);
+
+	const isTopLevelItemActive =
+		itemsWithIsActive.find((item) => item.isActive)?.href === activePath;
 
-	return <LinkList activePath={activePath} items={items} />;
-}
+	return (
+		<LinkList
+			activePath={activePath}
+			isListOpen={isTopLevelItemActive}
+			items={itemsWithIsActive}
+			subLevelVisible={subLevelVisible}
+		/>
+	);
+};
 
 type LinkListProps = {
 	activePath: string | undefined;
-	items: (SideNavProps['items'][number] & { isActive?: boolean })[];
+	isListOpen: boolean;
+	items: ItemWithIsActive[];
+	subLevelVisible: SideNavProps['subLevelVisible'];
 };
 
-function LinkList({ activePath, items }: LinkListProps) {
+const LinkList = ({
+	activePath,
+	isListOpen,
+	items,
+	subLevelVisible,
+}: LinkListProps) => {
+	const isOpen = useCallback(
+		(
+			items?: SideNavLinkListProps['items'] | ItemWithIsActive[],
+			isActive?: boolean
+		) =>
+			isActive ||
+			subLevelVisible === 'always' ||
+			hasSubLevelActiveItem(items, activePath),
+		[activePath, subLevelVisible]
+	);
+
 	return (
-		<SideNavGroup>
-			{items.map(({ isActive, items, ...item }, idx) => {
+		<SideNavUnorderedList isOpen={isOpen(items, isListOpen)}>
+			{items.map(({ isActive, items, ...item }) => {
+				const numberOfItems = items?.length || 0;
+				const hasNestedItemsIndicator =
+					numberOfItems > 0 && subLevelVisible === 'whenActive';
+
 				return (
-					<SideNavLink
-						key={idx}
-						isActive={isActive}
-						isCurrentPage={item.href === activePath}
-						{...item}
+					<SideNavListItem
+						isActive={isActive || hasSubLevelActiveItem(items, activePath)}
+						key={item.href}
 					>
+						<SideNavLink
+							hasNestedItemsIndicator={hasNestedItemsIndicator}
```

Comment:
Still have references to Nested

Reply:
- [ ] hasNestedItemsIndicator - > hasSubLevelItemsIndicator